aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -161,10 +161,76 @@ static void where_kernel_impl(TensorIterator &iter, ScalarType condition_type) {
   });
 }
 
+static void isposinf_kernel_for_bool_impl(TensorIterator& iter) {
+  if (c10::isIntegralType(iter.input_dtype(), /*include_bool=*/true)) {
+    AT_DISPATCH_INTEGRAL_TYPES_AND(at::ScalarType::Bool, iter.input_dtype(), "isposinf_cpu", [&]() {
+      cpu_kernel(iter, [](scalar_t a) -> bool { return false; });
+    });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, iter.input_dtype(), "isposinf_cpu", [&]() {
```

Comment:
This is interesting. It's dispatching on bloat16, but I thought std::numeric_limits didn't support bfloat16? For bfloat16 do you need a custom dispatch that instead calls `infinity` from here:

https://github.com/pytorch/pytorch/blob/1a0b95e7e46a056546a49f7f8b40aaf6ffa11dde/c10/util/BFloat16-inl.h#L249

?

Reply:
I found the namespace for `static constexpr c10::BFloat16 infinity()` is actually **std**.
https://github.com/pytorch/pytorch/blob/c0bfa45f9d26fb814a80f3053fbc550ff77d82c1/c10/util/BFloat16-inl.h#L203-L209

I added the `print` clauses to the `c10::BFloat16 infinity()` function in my local dev environment, to see whether the `c10::BFloat16 infinity()` is called when dealing with `bfloat16` tensors.
The log is outputted to the console so that my guess is the `std::numeric_limits<scalar_t>::infinity()` used in the code may not belong to the cpp standard.

Reply:
Aha! Thank you for reviewing and explaining this. That makes sense.